dcpy/lifecycle/package/assemble.py

```diff
@@ -96,4 +98,48 @@ def unzip_into_package(
             )
 
 
+def assemble_dataset_from_bytes(
+    dataset_metadata: md.Metadata,
+    *,
+    destination_id: str,
+    out_path: Path | None,
+):
+    out_path = out_path or ASSEMBLY_DIR / dataset_metadata.id
+    logger.info(f"Assembling dataset from BYTES. Writing to: {out_path}")
+    bytes.pull_destination_files(
+        out_path, dataset_metadata, destination_id, unpackage_zips=True
+    )
+
+    oti_data_dictionaries = [
+        f.file
+        for f in dataset_metadata.files
+        if f.file.type == oti_xlsx.OTI_METADATA_FILE_TYPE
+    ]
+    if len(oti_data_dictionaries) > 0:
+        # minor TODO: what to do if there are multiple oti data dictionaries specified?
```

Comment:
This is where maybe it'd be nice to expand some validation utils - just a function called `list.exactly_one` that gives you the single element of a list or throws an error.

In this case, do we want to throw an error? in the strange case that we query bytes and somehow get two

Reply:
Or sorry - this is based on what's defined in our metadata, not what's found on bytes, correct?

Might also be worth adding pydantic validation to the class if that's a real constraint. Happy to leave that for now though

Reply:
> Or sorry - this is based on what's defined in our metadata, not what's found on bytes, correct?

Yeah, what's in the metadata. I've changed this to just generate each OTI file listed, because it seems more consistent. (I don't think there's a real use case for it, but this is all going to change once we have other assets to generate)